### PR TITLE
fix(Core/Spell): Big Blizzard Bear

### PR DESCRIPTION
**Pull request by poszer**
Fix for mount Big Blizzard Bear

##### CHANGES PROPOSED:

- Correctly apply 60% & 100% speed for mounts based on riding skill
-  

###### ISSUES ADDRESSED:

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1896

##### TESTS PERFORMED:

Yes, builded without errors. Tested in game, now everything is fine.

##### HOW TO TEST THE CHANGES:

Read this issue : https://github.com/azerothcore/azerothcore-wotlk/issues/1896

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

### CODE REVIEW

File: src/server/scripts/Spells/spell_generic.cpp [hunk 1]
```diff
@@ -4374,7 +4374,10 @@ enum Mounts
 
     // Big Blizzard Bear
     SPELL_BIG_BLIZZARD_BEAR_60          = 58997,
-    SPELL_BIG_BLIZZARD_BEAR_100         = 58999
+    SPELL_BIG_BLIZZARD_BEAR_100         = 58999,
+    SPELL_BIG_BLIZZARD_BEAR_150         = 58999,
+    SPELL_BIG_BLIZZARD_BEAR_280         = 58999,
+    SPELL_BIG_BLIZZARD_BEAR_310         = 58999
```
Comment (Meltie2013) [inline src/server/scripts/Spells/spell_generic.cpp]: Only 60 & 100 exist for this mount. No need to double the same spell Id here.

File: src/server/scripts/Spells/spell_generic.cpp [hunk 2]
```diff
@@ -4935,7 +4938,7 @@ void AddSC_generic_spell_scripts()
     new spell_gen_disabled_above_63();
     new spell_gen_black_magic_enchant();
     new spell_gen_area_aura_select_players();
-    new spell_gen_mount("spell_big_blizzard_bear", 0, SPELL_BIG_BLIZZARD_BEAR_60, SPELL_BIG_BLIZZARD_BEAR_100, 0, 0);
+    new spell_gen_mount("spell_big_blizzard_bear", 0, SPELL_BIG_BLIZZARD_BEAR_60, SPELL_BIG_BLIZZARD_BEAR_100, SPELL_BIG_BLIZZARD_BEAR_150, SPELL_BIG_BLIZZARD_BEAR_280, SPELL_BIG_BLIZZARD_BEAR_310);
```
Comment (Meltie2013) [inline src/server/scripts/Spells/spell_generic.cpp]: This should work and remove the extra `0, 0` at the end.
`new spell_gen_mount("spell_big_blizzard_bear", 0, SPELL_BIG_BLIZZARD_BEAR_60, SPELL_BIG_BLIZZARD_BEAR_100);`

Comment (poszer) [reply src/server/scripts/Spells/spell_generic.cpp]: > This should work and remove the extra `0, 0` at the end.
> `new spell_gen_mount("spell_big_blizzard_bear", 0, SPELL_BIG_BLIZZARD_BEAR_60, SPELL_BIG_BLIZZARD_BEAR_100);`

Nope, doesn't work. Already tested like that, before opening this PR. 
1. Learning Apprentice riding -> Big Blizzard Bear **_(Inc speed by 60) - OK_**
2. Learning Journeyman riding ->  Big Blizzard Bear **_(Inc speed by 100) - OK_**
3. **_Learning Expert riding -> Issue (It revert it to slow speed)_**
4. **_Learning Artisan riding -> Same issue_**

File: src/server/scripts/Spells/spell_generic.cpp (hunk 1 above)
Comment (poszer) [reply src/server/scripts/Spells/spell_generic.cpp]: I aggre, but it doesn't work :smiley:

**Comment by Mindustry** [comment]
Tested #1905 on Windows 10 64-bit build. Closes #1896 
All changes came out successful. Well done quick fix, @poszer. Thanks to @rasm685p for reporting the issue. 
snipped

Tested without the fix
1. Upgrading to 100% riding speed while mounted on Big Blizzard Bear
2. Moving around at 100% after remounting

Result is the wrong speed buff being applied to the player. 
[Video](http://webmshare.com/play/OxMXy)

Tested with the fix
1. Upgrading to 100% riding speed while mounted on Big Blizzard Bear
2. Moving around at 100% after remounting

This patch fixes the issue and correctly applies the 100% speed buff. 
[Video](http://webmshare.com/play/QezRR)

**Comment by poszer** [comment]
> Tested #1905 on Windows 10 64-bit build. Closes #1896
> All changes came out successful. Well done quick fix, @poszer. Thanks to @rasm685p for reporting the issue.
> snipped
> 
> Tested without the fix
> 
> 1. Upgrading to 100% riding speed while mounted on Big Blizzard Bear
> 2. Moving around at 100% after remounting
> 
> Result is the wrong speed buff being applied to the player.
> [Video](http://webmshare.com/play/OxMXy)
> 
> Tested with the fix
> 
> 1. Upgrading to 100% riding speed while mounted on Big Blizzard Bear
> 2. Moving around at 100% after remounting
> 
> This patch fixes the issue and correctly applies the 100% speed buff.
> [Video](http://webmshare.com/play/QezRR)

Nice! Thank you for testing this :+1:

**Pull request merged**